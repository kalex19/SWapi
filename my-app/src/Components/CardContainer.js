import React, { Component } from 'react';
import './CardContainer.scss';
import ScrollText from './ScrollText.js';
// import ErrorMessage from './ErrorMessage.js';
// import Card from './Card.js';

const CardContainer = props => {
	//functionality or conitional rendering?
	//possibly functionality to pick random film

	return (
		<main>
			<ScrollText film={props.film} />
			{/* //displayed
      while other data is loading ? ? goes away and cards are displayed 
      < /main>/
      if no favorites.... < ErrorMessage / >
      if people btn....create < Card people = {
        this.props.people
      }
      if planets btn....create < Card planets = {
        this.props.planets
      }
      if vehicles btn....create < Card vehicles = {
        this.props.vehicle
      } */}
		</main>
	);
};

// CardContainer.propTypes = {
//   film: PropTypes.object,
//   people: PropTypes.array,
//   planets: PropTypes.array,
//   vehicles: PropTypes.array,
//   //removeIdea: PropTypes.func
// }

export default CardContainer;
