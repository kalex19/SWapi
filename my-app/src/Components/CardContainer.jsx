import React, { Component } from 'react';
import './CardContainer.scss';
import PropTypes from 'prop-types';
import ScrollText from './ScrollText';
import ErrorMessage from './ErrorMessage';
import Card from './Card';

const CardContainer = props => {
	let errorMessage;
	const cards = (
		<Card
			category={props.category}
			people={props.people}
			planets={props.planets}
			vehicles={props.vehicles}
			count={props.count}
			favoritesBtn={props.favoritesBtn}
			active={props.active}
		/>
	);

	if (props.category === null && props.count === 0) {
		errorMessage = <ErrorMessage />;
	}

	return (
		<main>
			{errorMessage}
			<ScrollText film={props.film} />
			{cards}
		</main>
	);
};

CardContainer.propTypes = {
	film: PropTypes.object,
	people: PropTypes.array,
	planets: PropTypes.array,
	vehicles: PropTypes.array,
	category: PropTypes.string,
	count: PropTypes.number,
	favoritesBtn: PropTypes.func,
	active: PropTypes.bool
};

export default CardContainer;
