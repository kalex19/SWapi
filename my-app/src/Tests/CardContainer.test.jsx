import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import CardContainer from '../Components/CardContainer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CardContainer', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<CardContainer />);
	});

	it('should mock snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('', () => {});

	it('', () => {});
});
