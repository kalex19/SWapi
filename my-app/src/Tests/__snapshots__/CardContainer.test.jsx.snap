// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should mock snapshot 1`] = `ShallowWrapper {}`;
